lib/core/src/display/navigation/navigator/events.rs

```diff
@@ -236,90 +228,96 @@ impl NavigatorEvents {
+        let listener = self.mouse_manager.on_wheel.add(move |event:&OnWheel| {
+            event.prevent_default();
             if let Some(data) = data.upgrade() {
-                if event.is_ctrl_pressed {
+                if event.ctrl_key() {
                     let position   = data.mouse_position();
                     let zoom_speed = data.zoom_speed();
-                    let zoom_event = ZoomEvent::from_mouse_wheel(event, position, zoom_speed);
+                    let amount     = event.delta_y() as f32;
+                    let zoom_event = ZoomEvent::new(position, amount, zoom_speed);
                     data.on_zoom(zoom_event);
                 } else {
-                    let pan_event = PanEvent::from_wheel_event(event);
+                    let x         =  event.delta_x() as f32;
+                    let y         = -event.delta_y() as f32;
+                    let movement  = Vector2::new(x,y);
+                    let pan_event = PanEvent::new(movement);
                     data.on_pan(pan_event);
                 }
             }
-        })?;
+        });
         self.wheel_zoom = Some(listener);
         Ok(())
     }
 
     fn initialize_mouse_start_event(&mut self) -> Result<()> {
         let data     = Rc::downgrade(&self.data);
-        let listener = self.mouse_manager.add_mouse_down_callback(move |event:MouseClickEvent| {
+        let listener = self.mouse_manager.on_down.add(move |event:&OnDown| {
             if let Some(data) = data.upgrade() {
-                match event.button {
-                    MouseButton::MIDDLE => {
+                match event.button() {
+                    button::MiddleButton => {
                         data.set_movement_type(Some(MovementType::Pan))
                     },
-                    MouseButton::RIGHT => {
-                        let focus = event.position;
+                    button::SecondaryButton => {
+                        let focus = Vector2::new(event.offset_x() as f32, event.offset_y() as f32);
                         data.set_movement_type(Some(MovementType::Zoom { focus }))
                     },
                     _ => ()
                 }
             }
-        })?;
+        });
         self.mouse_down = Some(listener);
         Ok(())
     }
 
-    fn disable_context_menu(&mut self) -> Result<()> {
-        self.disable_context_menu = Some(self.mouse_manager.disable_context_menu()?);
+    fn disable_context_menu(&mut self, target:&EventTarget) -> Result<()> {
+        self.disable_context_menu = Some(ignore_context_menu(target)?);
         Ok(())
     }
 
     fn initialize_mouse_end_event(&mut self) -> Result<()> {
-        let data         = Rc::downgrade(&self.data);
-        let closure      = move |_| {
+        let data              = Rc::downgrade(&self.data);
+        let listener          = self.mouse_manager.on_up.add(move |_:&OnUp| {
             if let Some(data) = data.upgrade() {
```

Comment:
a minor thing - do not align `=` if they are in different blocks . So here, the last `=` is in a new block `{...}` , so the first 2 `=` sohuld be aligned with themselves but not with the last one.